addons/binding/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/handler/SmartthingsBridgeHandler.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smartthings.handler;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.smartthings.config.SmartthingsBridgeConfig;
+import org.openhab.binding.smartthings.internal.SmartthingsHttpClient;
+//import org.eclipse.equinox.event.
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SmartthingsBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bob Raker - Initial contribution
+ */
+public class SmartthingsBridgeHandler extends BaseBridgeHandler {
+    private Logger logger = LoggerFactory.getLogger(SmartthingsBridgeHandler.class);
+
+    private SmartthingsBridgeConfig config;
+
+    private SmartthingsHttpClient httpClient;
+
+    public SmartthingsBridgeHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Commands are handled by the "Things" which include Cm11aSwitchHandler and Cm11aLampHandler
+    }
+
+    @Override
+    public void initialize() {
+        // Get config data and validate
+        config = getThing().getConfiguration().as(SmartthingsBridgeConfig.class);
+        if (!validateConfig(this.config)) {
+            return;
+        }
+
+        // Initialize the Smartthings http client
+        try {
+            httpClient = new SmartthingsHttpClient(config.smartthingsIp, config.smartthingsPort);
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Failed to start Smartthings Communications services because: " + e.getMessage());
+            return;
+        }
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Smartthings Handler disposed.");
+
+        httpClient.stopHttpClient();
+
+        super.dispose();
+    }
+
+    private boolean validateConfig(SmartthingsBridgeConfig config) {
```

Comment:
Extend from `ConfigStatusThingHandler`. That allows you to validate and return a more explicit error to the framework and user-interfaces.